pkg/server/register.go

```diff
@@ -395,6 +403,51 @@ func (i *InventoryServer) serveLoop(conn *websocket.Conn, inventory *elementalv1
 	}
 }
 
+func (i *InventoryServer) handleGet(conn *websocket.Conn, protoVersion register.MessageType, inventory *elementalv1.MachineInventory, registration *elementalv1.MachineRegistration) error {
+	var err error
+
+	inventory, err = i.commitMachineInventory(inventory)
+	if err != nil {
+		if writeErr := i.writeError(conn, protoVersion, register.NewErrorMessage(err)); writeErr != nil {
+			logrus.Errorf("Error reporting back error to client: %v", writeErr)
+		}
+
+		return err
+	}
+
+	if err := i.writeMachineInventoryCloudConfig(conn, protoVersion, inventory, registration); err != nil {
+		if writeErr := i.writeError(conn, protoVersion, register.NewErrorMessage(err)); writeErr != nil {
+			logrus.Errorf("Error reporting back error to client: %v", writeErr)
+		}
+
+		return fmt.Errorf("failed sending elemental cloud config: %w", err)
+	}
+
+	logrus.Debug("Elemental cloud config sent")
+
+	if protoVersion == register.MsgUndefined {
+		logrus.Warn("Detected old elemental-register client: cloud-config data may not be applied correctly")
+	}
+
+	return nil
+}
+
+// writeError reports back an error to the client if the negotiated protocol
+// version supports it.
+func (i *InventoryServer) writeError(conn *websocket.Conn, protoVersion register.MessageType, errorMessage register.ErrorMessage) error {
```

Comment:
Nice, the writeError function!
nit: we don't need the InventoryServer receiver argument here
```suggestion
func writeError(conn *websocket.Conn, protoVersion register.MessageType, errorMessage register.ErrorMessage) error {
```

Reply:
Nice, removing this in a new commit :+1: